src/modules/ekf2/EKF2Selector.cpp

```diff
@@ -686,23 +694,31 @@ void EKF2Selector::Run()
 	}
 
 	if (_param_ekf2_gnss_denied.get() == 1) {
-		_desired_pos_est_mode = estimator_status_s::POS_EST_MODE_VISION_DENIED;
-	}
+		if (_param_ekf2_sel_gnss_denied.get() != 1 || _latch_use_gnss) {
+			_desired_pos_est_mode = estimator_status_s::POS_EST_MODE_VISION_DENIED;
```

Comment:
While reading this line I was at first a bit confused here, because I read

`POS_EST_MODE_VISION_DENIED` as
"Use vision for position, we are GNSS denied"

So it is an indication that the name could be improved. (But not critical, I understood pretty quickly)